File: LogicaFor.js
var edad = 0, estadoCivil = 0, estatura = 0, sexo = 0, promedioEdad = 0, promedioEstatura = 0, porcentajePersonas = 0, 
contadorPersonas = 0, acomuladorEdad = 0, acomuladorEstatura = 0, cantidadPersonas = 2;

for (let numeroPersona = 1; numeroPersona <= cantidadPersonas; numeroPersona++) {
    edad = parseFloat(prompt("Ingrese la edad de la persona " + numeroPersona));
    estadoCivil = parseFloat (prompt("Ingrese el estado civil de la persona " + numeroPersona + " 1. Soltero   2. Casado"));
    estatura = parseFloat (prompt("Ingrese la estatura de la persona " + numeroPersona + " en cms"));
    sexo = parseFloat (prompt("Ingrese el sexo de la persona " + numeroPersona + " 1. Hombre     2. Mujer"));

    if ((edad >= 18)&&(estadoCivil==1)&&(estatura>170)&&(sexo=1)){
    contadorPersonas = contadorPersonas + 1;
    acomuladorEdad = acomuladorEdad + edad;
    acomuladorEstatura = acomuladorEstatura + estatura;
    }
}

promedioEdad = acomuladorEdad / contadorPersonas;
promedioEstatura = acomuladorEstatura / contadorPersonas;
porcentajePersonas = (contadorPersonas/cantidadPersonas)*100;

alert("El promedio de edad de las personas que cumplen es: " + promedioEdad + " años ");
alert("El promedio de estatura de las personas que cumplen es: " + promedioEstatura + "centimetros");
alert("El porcentaje de personas que cumplen con la condición es del: " + porcentajePersonas + "%");

File: LogicaFor.java
public class LogicaFor{
int edad = 0, estadoCivil = 0, estatura = 0, sexo = 0, contadorPersonas, acomuladorEdad = 0, acomuladorEstatura = 0,  numeroPersona = 1;
int cantidadPersonas;


public LogicaFor(int edad, int estadoCivil, int estatura, int sexo, int contadorPersonas, int acomuladorEdad,
                int acomuladorEstatura, int numeroPersona, int cantidadPersonas) {
        this.edad = edad;
        this.estadoCivil = estadoCivil;
        this.estatura = estatura;
        this.sexo = sexo;
        this.contadorPersonas = contadorPersonas;
        this.acomuladorEdad = acomuladorEdad;
        this.acomuladorEstatura = acomuladorEstatura;
        this.numeroPersona = numeroPersona;
        this.cantidadPersonas = cantidadPersonas;
}


public void calcularPromedio(){

double promedioEdad = acomuladorEdad / contadorPersonas;
double promedioEstatura = acomuladorEstatura / contadorPersonas;
double porcentajePersonas = (contadorPersonas/cantidadPersonas)*100;

System.out.println("El promedio de edad de las personas que cumplen es: " + promedioEdad + " años ");
System.out.println("El promedio de estatura de las personas que cumplen es: " + promedioEstatura + " centimetros ");
System.out.println("El porcentaje de personas que cumplen con la condición es del: " + porcentajePersonas + " % ");
 

 }
}